minard/views.py

```diff
@@ -1576,3 +1579,69 @@ def _dropout_detail_n100(run_number):
 @app.route("/dropout/_dropout_detail/N20/<int:run_number>")
 def _dropout_detail_n20(run_number):
     return dropout.get_details(run_number, 2)
+
+@app.route('/standard_runs/<uuid>', methods=['GET', 'POST'])
+@app.route("/standard_runs/")
+@app.route("/standard_runs")
+def standard_runs(uuid=None):
+    if uuid is None:
+        values = None
+        try:
+            values = sr.get_standard_runs()
+        except sr.couchdb.http.socket.error:
+            flash("Error connecting to database, could not retrieve standard runs", "danger")
+        return render_template('standard_runs.html', values=values)
+    if request.method =='POST':
+        updater = request.form.get("name")
+        info = request.form.get("info")
+        passw = request.form.get("password")
+        new_values = [x[:-8] for x in request.form.keys() if x[-8:] == "_replace"]
+        new_values = {x:request.form[x+"_value"] for x in new_values}
+        # Convert values from strings to floats/integers/bools if appropriate
+        new_values = {k:True if v.lower() == 'true' else v for k,v in new_values.iteritems() if v}
+        new_values = {k:False if v.lower() == 'false' else v for k,v in new_values.iteritems()}
+        new_values = {k:float(v) if v.isdigit() else v for k,v in new_values.iteritems()}
+        new_values = {k:int(v) if type(v)==float and v.is_integer() else v for k,v in new_values.iteritems()}
```

Comment:
I think it would be good to have some server side verification of all of the values here. Do I understand that currently I could add "foo" as the trigger mask and it would get saved as a string? I'm not sure what Orca does in that case. You can use wtforms to do this validation. It's used by the channel database here: https://github.com/snoplus/minard/blob/926f1b74c5a247f8dadccb4de8cb1c55e8148845/minard/channeldb.py#L7 and here: https://github.com/snoplus/minard/blob/master/minard/views.py#L372.

Reply:
I agree that verification would be good, but I didn't want to get to bogged down figuring out whats valid input for all ~50 standard run settings. I figured making it expert only  might be good enough because the most common use case for this (I suspect) will be copy/pasting settings from one run to another. That being said I can take a crack at adding validation this weekend, it may not be as time consuming as I expect.

 And I looked into using WTFForms, the issue I ran into with that is (I think) it requires you know the fields you want ahead of time, but since couch is so amorphous the fields can change pretty easily. I could just implement a set of forms for the current standard run settings, but that will then mean we have to update this webpage whenever we add/remove some standard run setting. Maybe there's some way to dynamically create forms, if so that'd be the best of both worlds.